### PR TITLE
Contest extra_options and artMeta cleanup

### PR DESCRIPTION
**Pull request by nekodex**
Renames `shape` to `thumbnail_shape` (within `extra_options`) and standardizes the derived attributes.
Also stops including the unused `artMeta` for non-art contests, plus general cleanups.

cc @Ephemeralis - be aware that once this is merged, you should start using `thumbnail_shape` instead of `shape` when making contests

### CODE REVIEW

**Comment by Ephemeralis** [comment]
Will old ones need to be migrated over to not break the listings, or will they be fine?

**Comment by nekodex** [comment]
Existing ones will automatically be migrated as part of this PR, so you don't need to worry about that.

File: resources/assets/coffee/react/contest-voting/art-entry.coffee [hunk 1]
```diff
@@ -45,15 +45,20 @@ export class ArtEntry extends React.Component
...
+    divClasses = 'contest-art-entry'
+    divClasses += " contest-art-entry--#{thumbnailShape}" if thumbnailShape
 
-    div style: { backgroundImage: osu.urlPresence(@props.entry.artMeta.thumb) }, className: _.compact(divClasses).join(' '),
+    if showVotes
+      divClasses += ' contest-art-entry--result'
+      if top3
+        divClasses += " contest-art-entry--placed contest-art-entry--placed-#{place}"
+      else
+        divClasses += ' contest-art-entry--smaller'
```
Comment (notbakaneko) [inline resources/assets/coffee/react/contest-voting/art-entry.coffee:56]: there's `classWithModifiers` in `utils/css` for enabling modifiers with `true`/`false` now

File: resources/assets/coffee/react/contest-voting/art-entry.coffee [hunk 2]
```diff
@@ -45,15 +45,20 @@ export class ArtEntry extends React.Component
...
 
-    divClasses = [
-      'contest-art-entry',
-      'contest-art-entry--result' if showVotes,
-      "contest-art-entry--placed contest-art-entry--placed-#{place}" if showVotes && top3,
-      'contest-art-entry--smaller' if showVotes && !top3,
-      "contest-art-entry--#{shape}" if shape
-    ]
+    divClasses = 'contest-art-entry'
+    divClasses += " contest-art-entry--#{thumbnailShape}" if thumbnailShape
```
Comment (notbakaneko) [inline resources/assets/coffee/react/contest-voting/art-entry.coffee:49]: really should be `if thumbnailShape?`

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: resources/assets/coffee/react/contest-voting/art-entry.coffee (hunk 1 above)
Comment (nekodex) [reply resources/assets/coffee/react/contest-voting/art-entry.coffee]: kinda not sure if `classWithModifiers` actually improves readability here... 🤔

File: resources/assets/coffee/react/contest-voting/art-entry.coffee [hunk 3]
```diff
@@ -45,15 +48,15 @@ export class ArtEntry extends React.Component
...
-      'contest-art-entry--smaller' if showVotes && !top3,
-      "contest-art-entry--#{shape}" if shape
-    ]
-
-    div style: { backgroundImage: osu.urlPresence(@props.entry.artMeta.thumb) }, className: _.compact(divClasses).join(' '),
+    div
+      style:
+        backgroundImage: osu.urlPresence(@props.entry.thumbnail)
+      className: classWithModifiers bn,
+        "#{thumbnailShape}": thumbnailShape?
```
Comment (notbakaneko) [inline resources/assets/coffee/react/contest-voting/art-entry.coffee:55]: `undefined: false`, `${shape}: true`, not sure if unintentional or genius 🤨

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**